### ISSUE TITLE
[Proposal] Add support for padding/masking to the attention computation (Minor feature request)

### ISSUE THREAD

**Issue by neelnanda-io** [neelnanda-io/TransformerLens#240]
### Proposal 

Models like BERT allow padding to the attention pattern, where the attention pattern is always zero for <PAD> tokens, ie the model cannot look at those tokens. This can also be done by explicitly passing in a Boolean attention mask, for models which do not have a separate PAD and END_OF_TEXT/START_OF_TEXT (EOS/BOS) tokens like GPT-2. 

### Motivation

This is mildly useful to have for even GPT-style models, as it means that if you eg input a batch of prompts of different lengths, they can be padded on the left, so that `logits[:, -1, :]` gets you the final token of each prompt, rather than logits[:, -1, :] being meaningless for shorter prompts.

I particularly want this to be part of exploratory analysis demo, so that when people input individual prompts it's harder to shoot yourself in the foot by having them be of different lengths. 

### Alternatives

The main alternative is just to do padding manually, or massage your prompts to all have the same length. 

### Additional context

This would also need to change absolute positional embeddings to use the attention mask, so that the first real token in each prompt gets the 0th positional embedding (it'll break things otherwise). I would implement this by having an attention mask object that gets optionally created by `to_tokens` if you give it a `left_pad=True` option, or possibly have a `to_tokens_left_pad` option, and which gets passed to `pos_embed` and to `attn`. Relative positional embedding schemes *should* be unaffected. 

This isn't that big a deal, just a mildly nice feature to have.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/MiniGrid/issues) in the repo (**required**)
